src/wallet/rpcwallet.cpp

```diff
@@ -3773,23 +3778,39 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
 
     CScript scriptPubKey = GetScriptForDestination(dest);
     ret.pushKV("scriptPubKey", HexStr(scriptPubKey.begin(), scriptPubKey.end()));
+
     const SigningProvider* provider = pwallet->GetSigningProvider(scriptPubKey);
 
     isminetype mine = pwallet->IsMine(dest);
     ret.pushKV("ismine", bool(mine & ISMINE_SPENDABLE));
+
     bool solvable = provider && IsSolvable(*provider, scriptPubKey);
     ret.pushKV("solvable", solvable);
+
     if (solvable) {
        ret.pushKV("desc", InferDescriptor(scriptPubKey, *provider)->ToString());
     }
+
     ret.pushKV("iswatchonly", bool(mine & ISMINE_WATCH_ONLY));
+
+    // Return DescribeWalletAddress fields.
+    // Always returned: isscript, ischange, iswitness.
+    // Optional: witness_version, witness_program, script, hex, pubkeys (array),
+    // sigsrequired, pubkey, embedded, iscompressed.
     UniValue detail = DescribeWalletAddress(pwallet, dest);
     ret.pushKVs(detail);
+
+    // Return label field if existing. Currently only one label can be
+    // associated with an address, so the label should be equivalent to the
+    // value of the name key/value pair in the labels hash array below.
     if (pwallet->mapAddressBook.count(dest)) {
         ret.pushKV("label", pwallet->mapAddressBook[dest].name);
     }
+
     ret.pushKV("ischange", pwallet->IsChange(scriptPubKey));
 
+    // Fetch KeyMetadata, if present, for the timestamp, hdkeypath, hdseedid,
```

Comment:
nit: Seems redundant. The code below is very easy to read.

Reply:
ACK, will update #17578 with this.

Reply:
Removed in #17578.